src/libxtp/threecenter_gwbse.cc

```diff
@@ -150,6 +154,19 @@ void TCMatrix_gwbse::FillBlock(std::vector<Eigen::MatrixXd>& block,
       }
     }  // gamma-loop
   }    // alpha-loop
+#if defined(USE_GPU)
```

Comment:
why copy the dftn and dftm to the gpu for every tensor instead of just once?

Reply:
Because this is a naive implementation :). I need to optimize the copies!

Reply:
okay... cool go ahead

Reply:
Actually, in that `#if defined` I am copying only once the `dftn` and `dftm` matrices. That is the reason that I use the `unordered_map`, to keep track of the already allocated matrices.
See the documentation of `triple_tensor_product` for further information.

Reply:
I just copy once, then I iterate over the tensor copy each submatrix to the GPU